### PR TITLE
Add Graph Explorer

### PR DESCRIPTION
**Pull request by NullSoldier**
A new graph explorer that helps debug the chain structure

Screen Shot 2021-05-04 at 4 41 44 PM

### CODE REVIEW

File: ironfish-graph-explorer/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Graph Explorer
+
+This project is used to visually explore the blockchain for debugging purposese. It's not the block explorer explorer.ironfish.network which is located at https://github.com/iron-fish/block-explorer
+
+![Screenshot](./screenshot.png)
+
+### How to Use
+
+Right now, data is pulled manually from src/data.json which is exported by `ironfish chain:export`. Down the line, we want to serve this live from the node itsel, or a data source such as an API that can serve the data real time.
```
Comment (dguenther) [inline ironfish-graph-explorer/README.md:9]:
```suggestion
Right now, data is pulled manually from src/data.json which is exported by `ironfish chain:export`. Down the line, we want to serve this live from the node itself, or a data source such as an API that can serve the data real time.
```

File: ironfish-graph-explorer/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Graph Explorer
+
+This project is used to visually explore the blockchain for debugging purposese. It's not the block explorer explorer.ironfish.network which is located at https://github.com/iron-fish/block-explorer
```
Comment (dguenther) [inline ironfish-graph-explorer/README.md:3]:
```suggestion
This project is used to visually explore the blockchain for debugging purposes. It's not the block explorer (https://explorer.ironfish.network), which is located at https://github.com/iron-fish/block-explorer
```

File: ironfish-cli/src/commands/chain/export.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  increment: VoidFunction
+}
+
+function parseNumber(input: string): number | null {
+  const parsed = Number(input)
+  return isNaN(parsed) ? null : parsed
+}
+
+export default class Export extends IronfishCommand {
+  static description = 'Export part of the chain database to JSONš'
```
Comment (dguenther) [inline ironfish-cli/src/commands/chain/export.ts:26]:
```suggestion
  static description = 'Export part of the chain database to JSON'
```

File: ironfish-cli/src/commands/chain/export.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+      parse: parseNumber,
+      default: Number(GENESIS_BLOCK_SEQUENCE),
+      required: false,
+      description: 'the sequence to start at inclusive. (genesis block is 1)',
+    },
+    {
+      name: 'stop',
+      parse: parseNumber,
+      required: false,
+      description: 'the sequence to end at inclusive.',
```
Comment (dguenther) [inline ironfish-cli/src/commands/chain/export.ts:51]:
```suggestion
      description: 'the sequence to end at (inclusive)',
```

File: ironfish-cli/src/commands/chain/export.ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    }),
+  }
+
+  static args = [
+    {
+      name: 'start',
+      parse: parseNumber,
+      default: Number(GENESIS_BLOCK_SEQUENCE),
+      required: false,
+      description: 'the sequence to start at inclusive. (genesis block is 1)',
```
Comment (dguenther) [inline ironfish-cli/src/commands/chain/export.ts:45]:
```suggestion
      description: 'the sequence to start at (inclusive, genesis block is 1)',
```

File: ironfish-graph-explorer/src/data.json [hunk 6]
```diff
@@ -0,0 +1,155 @@
+[
+  {
+    "hash": "a1ba91bd54fcfe97d0df8fe1f66b6217197b96cd52db97fad58b275e164600ab",
```
Comment (dguenther) [inline ironfish-graph-explorer/src/data.json:3]: Should this be deleted?

File: ironfish-graph-explorer/package.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "graphviewer",
```
Comment (dguenther) [inline ironfish-graph-explorer/package.json:2]: not sure if this is worth changing or not
```suggestion
  "name": "ironfish-graph-explorer",
```

File: ironfish-graph-explorer/src/data.json (hunk 6 above)
Comment (NullSoldier) [reply ironfish-graph-explorer/src/data.json:3]: Let's keep it as a small demo so they don't have to export to see it working

**Review by dguenther — APPROVED** [review]

**Pull request merged**